src/Analyzers/VisualBasic/Analyzers/ParenthesesStyle/VisualBasicParenthesesStyleDiagnosticAnalyzer.vb

```diff
@@ -0,0 +1,62 @@
+Imports Microsoft.CodeAnalysis.Options
+Imports Microsoft.CodeAnalysis.VisualBasic.CodeStyle
+Imports Microsoft.CodeAnalysis.Operations
+
+Namespace Microsoft.CodeAnalysis.VisualBasic.ParenthesesStyle
+    <DiagnosticAnalyzer(LanguageNames.VisualBasic)>
+    Friend Class VisualBasicParenthesesStyleDiagnosticAnalyzer
+        Inherits AbstractBuiltInCodeStyleDiagnosticAnalyzer
+
+        Private Shared ReadOnly s_descriptorForInvocations As New DiagnosticDescriptor(IDEDiagnosticIds.ParenthesesStyleDiagnosticId,
+                                                                              New LocalizableResourceString(NameOf(VisualBasicAnalyzersResources.Add_invocation_parentheses), VisualBasicAnalyzersResources.ResourceManager, GetType(VisualBasicAnalyzersResources)),
+                                                                              New LocalizableResourceString(NameOf(VisualBasicAnalyzersResources.Add_invocation_parentheses), VisualBasicAnalyzersResources.ResourceManager, GetType(VisualBasicAnalyzersResources)),
+                                                                              category:="Not sure what to put here.",
+                                                                              defaultSeverity:=DiagnosticSeverity.Hidden,
+                                                                              isEnabledByDefault:=False)
+
+        Private Shared ReadOnly s_descriptorForProperties As New DiagnosticDescriptor(IDEDiagnosticIds.ParenthesesStyleDiagnosticId,
+                                                                              New LocalizableResourceString(NameOf(VisualBasicAnalyzersResources.Remove_property_parentheses), VisualBasicAnalyzersResources.ResourceManager, GetType(VisualBasicAnalyzersResources)),
+                                                                              New LocalizableResourceString(NameOf(VisualBasicAnalyzersResources.Remove_property_parentheses), VisualBasicAnalyzersResources.ResourceManager, GetType(VisualBasicAnalyzersResources)),
+                                                                              category:="Not sure what to put here.",
+                                                                              defaultSeverity:=DiagnosticSeverity.Hidden,
+                                                                              isEnabledByDefault:=False)
+
+        Public Sub New()
+            MyBase.New(ImmutableDictionary(Of DiagnosticDescriptor, ILanguageSpecificOption).Empty.Add(
+                       s_descriptorForInvocations, VisualBasicCodeStyleOptions.ParenthesesForInvocations).Add(
+                       s_descriptorForProperties, VisualBasicCodeStyleOptions.ParenthesesForProperties), LanguageNames.VisualBasic)
+        End Sub
+
+        Protected Overrides Sub InitializeWorker(context As AnalysisContext)
+            ' How to retrieve the editorconfig value to decide whether to proceed or not?
```

Comment:
you don't need to do this here.  this is the registration point.  you can't check editorconfig here as you won't know what file-location you're currently in.